### DISCUSSION TITLE
Impl.ts 파일 분리

### DISCUSSION THREAD

**Discussion by nyj001012** [jiphyeonjeon-42/backend#600] [Ideas]
Impl.ts에 서비스 내용과 컨트롤러 내용이 같이 있는데, 폴더 구조를
- service
- service/impl.ts

- controller
- controller/impl.ts

처럼 형식을 바꿔보는 건 어떨까요?

**Comment by Yonaim** [comment]
동의합니다. 레이어드 아키텍처 패턴을 사용하고 있는데 impl.ts의 존재 때문에 패턴을 사용하는 의미가 없어지는 것 같아요.

**Reply by scarf005** [reply]
혹시 어떤 부분에서 그렇게 생각하셨는지 여쭤볼 수 있을까요? 

현재 구조에서는
- [`controller.ts`](https://github.com/jiphyeonjeon-42/backend/blob/cc1ec7b7e569302abb46e1a6eed7ac94c03f0e48/backend/src/v2/reviews/controller.ts): 컨트롤러 구현 (의존성: 서비스)
- [`service/`](https://github.com/jiphyeonjeon-42/backend/blob/cc1ec7b7e569302abb46e1a6eed7ac94c03f0e48/backend/src/v2/reviews/service/service.ts): 서비스 구현 (의존성: Typeorm repository)
- [impl.ts](https://github.com/jiphyeonjeon-42/backend/blob/cc1ec7b7e569302abb46e1a6eed7ac94c03f0e48/backend/src/v2/reviews/impl.ts): 컨트롤러와 서비스에 의존성을 '주입'해주는 파일

식으로 되어 있는데, [implReviewService]와 [implReviewController]처럼, 컨트롤러와 서비스를 다루는 코드가 같은 파일에 있어 레이어드 아키텍처 패턴이 지켜지지 않는다는 말씀이실까요?

[implReviewService]: https://github.com/jiphyeonjeon-42/backend/blob/cc1ec7b7e569302abb46e1a6eed7ac94c03f0e48/backend/src/v2/reviews/impl.ts#L19-L27

[implReviewController]: https://github.com/jiphyeonjeon-42/backend/blob/cc1ec7b7e569302abb46e1a6eed7ac94c03f0e48/backend/src/v2/reviews/impl.ts#L29-L34

**Reply by Yonaim** [reply]
레이어드 아키텍처가 지켜지지 않다기보다는 밑의 yena님 말씀처럼 헷갈린다는 의미였습니다ㅎㅎ 해결된 것 같군요

**Reply by scarf005** [reply]
넵 질문 감사합니당

**Comment by scarf005** [comment]
좋은 아이디어입니다. 그렇다면
`backend/src/v2/reviews/impl.ts`가 `service/impl.ts`와 `controller/impl.ts`의 구현을 import하는 식으로 볼 수 있을까요?

**Reply by nyj001012** [reply]
넵, 서비스랑 컨트롤러에 대한 impl 부분이 한 파일에 섞여있다 보니까 헷갈리더라구요

**Reply by scarf005** [reply]
그렇다면 이슈로 변경해 작업하겠습니다
